src/mgmt_conn.c

```diff
@@ -1042,16 +1042,25 @@ uzfs_zvol_execute_async_command(void *arg)
 		break;
 	case ZVOL_OPCODE_RESIZE:
 		volsize = *(uint64_t *)async_task->payload;
-		rc = zvol_check_volsize(volsize,
-		    zinfo->main_zv->zv_volblocksize);
-		if (rc == 0) {
-			rc = zvol_update_volsize(volsize,
-			    zinfo->main_zv->zv_objset);
-			if (rc == 0)
-				zvol_size_changed(zinfo->main_zv, volsize);
+		if (uzfs_zvol_get_status(zinfo->main_zv) ==
+		    ZVOL_STATUS_HEALTHY) {
+			// TODO: Is assertion with snapshot and clone
+			// data set is required here?
```

Comment:
Take a lock of `zinfo->main_zv->rebuild_mtx` before checking the replica status.

Reply:
Took the lock